### PR TITLE
Fix zenith calculation code so that all the basic inputs successfully calculate zenith delays

### PR DESCRIPTION
**Pull request by jlmaurer**
- This addresses #37
- test_scenario_1 and test_scenario_2 should complete successfully now, these are both end-to-end units tests
- Note: test_scenario_1 and test_scenario_2 test will test against the test data set, but the test data set has not been validated. In other words, these unit tests will test for changing values against what the code has been providing, but not the absolute accuracy.
- chunking is still based on cpu count but adds a safety check; if the chunksize would be greater or less than a threshold it takes the threshold
- chunking is fixed for station_file (i.e. 1D) and height_level (i.e. 3D) query points
- Added unit tests
- This is a new version of #69

### CODE REVIEW

**Comment by jlmaurer** [comment]
Fixed conflicts, all unit tests (finally!) passing.

**Comment by dbekaert** [comment]
> Can you please fix the formatting? I already comment on a lot of issues on the previous PR, so just go look at the comments on that one.
> 
>     * Closing braces are not aligned properly
> 
>     * Indentation is not a multiple of 4

@Askaholic is there a test that we can run which would fail, similar like circleCI?
Are you aware of any applications?

**Comment by Askaholic** [comment]
Yes, Codacy and Codefactor are ones that I've used before, but there are literally dozens of services out there that will all do basically the same thing, for example: https://www.guru99.com/code-review-tools.html 

You could even set up a travis or circleCI built do run a linter if you wanted.

**Comment by dbekaert** [comment]
@Askaholic we have codacy but i am not able to get it to work in our repo here for some reason. I think the repo needs to be part of an org, otherwize it only check if i make the commit. Can you confirm?

**Comment by Askaholic** [comment]
Yea Codacy does permissions in a strange way, I'm not sure if there's a way to get around it. I haven't set it up myself, but I did see the person who set it up on my open source project complain about their wacky permissions.

File: test/test_interpolator.py [hunk 1]
```diff
@@ -477,7 +480,8 @@ def test_3d_basic():
     ys = np.array([0, 1])
     zs = np.array([0, 1])
 
-    values = (lambda x, y, z: x + y + z)(*np.meshgrid(xs, ys, zs, indexing="ij", sparse=True))
+    values = (lambda x, y, z: x + y + z)(*
+                                         np.meshgrid(xs, ys, zs, indexing="ij", sparse=True))
```
Comment (Askaholic) [inline test/test_interpolator.py:484]: Not sure about this one, I'd probably go with
```suggestion
    values = (lambda x, y, z: x + y + z)(
        *np.meshgrid(xs, ys, zs, indexing="ij", sparse=True)
    )
```

File: test/test_interpolator.py [hunk 2]
```diff
@@ -663,7 +667,8 @@ def test_4d_basic():
     zs = np.array([0, 1])
     ws = np.array([0, 1])
 
-    values = (lambda x, y, z, w: x + y + z + w)(*np.meshgrid(xs, ys, zs, ws, indexing="ij", sparse=True))
+    values = (lambda x, y, z, w: x + y + z + w)(*
```
Comment (Askaholic) [inline test/test_interpolator.py:670]: same

File: test/test_util.py [hunk 3]
```diff
@@ -146,13 +194,13 @@ def test_makeDelayFileNames():
         (
             "dir/model_name_wet_01_02_03_ztd.h5",
             "dir/model_name_hydro_01_02_03_ztd.h5"
-        )
+    )
```
Comment (Askaholic) [inline test/test_util.py:197]: It was good before.

File: tools/RAiDER/models/allowed.py [hunk 4]
```diff
@@ -11,7 +11,8 @@ def AllowedModels():
       'WRF',
       'HRRR',
       'GMAO',
-      'HDF5']
+      'HDF5'
+      ]
```
Comment (Askaholic) [inline tools/RAiDER/models/allowed.py:15]: IMO the indentation should be fixed so it's a multiple of 4

File: test/test_scenario_1.py [hunk 5]
```diff
@@ -55,8 +62,11 @@ def test_tropo_delay(tmp_path):
         # get the results
         wet = gdal_open(wet_file)
         hydro = gdal_open(hydro_file)
-        true_wet = gdal_open(os.path.join(SCENARIO_DIR, "ERA5_wet_true.envi"))
-        true_hydro = gdal_open(os.path.join(SCENARIO_DIR, "ERA5_hydro_true.envi"))
+        true_wet = gdal_open(os.path.join(
+            SCENARIO_DIR, "wet.envi"), userNDV=0.)
+        true_hydro = gdal_open(os.path.join(
+            SCENARIO_DIR, "hydro.envi"), userNDV=0.)
+        print(true_wet)
```
Comment (Askaholic) [inline test/test_scenario_1.py:69]: delete debug output
```suggestion
```

File: tools/RAiDER/utilFcns.py [hunk 6]
```diff
@@ -67,13 +65,17 @@ def gdal_open(fname, returnProj=False):
...
-        d = b.ReadAsArray()
-        try:
-            ndv = b.GetNoDataValue()
-            d[d == ndv] = np.nan
-        except:
-            pass
-        val.append(d)
+        data = b.ReadAsArray()
+        if userNDV is not None:
+            print('Using user-supplied NoDataValue')
```
Comment (Askaholic) [inline tools/RAiDER/utilFcns.py:70]: use the logger

File: tools/RAiDER/utilFcns.py [hunk 7]
```diff
@@ -455,22 +456,28 @@ def writePnts2HDF5(lats, lons, hgts, los, outName='testx.h5', chunkSize=None):
...
     os.makedirs(os.path.abspath(os.path.dirname(outName)), exist_ok=True)
 
+    if chunkSize is None:
+        minChunkSize = 100
+        maxChunkSize = 10000
+        cpu_count = mp.cpu_count()
+        chunkSize = tuple(max(min(maxChunkSize, s//cpu_count), min(s, minChunkSize)) for s in in_shape)
+
+    if verbose:
+        print('Chunk size is {}'.format(chunkSize))
```
Comment (Askaholic) [inline tools/RAiDER/utilFcns.py:466]: use the logger

Comment (jlmaurer) [reply tools/RAiDER/utilFcns.py]: @Askaholic fixed.

File: tools/RAiDER/utilFcns.py (hunk 6 above)
Comment (jlmaurer) [reply tools/RAiDER/utilFcns.py]: @Askaholic fixed.

File: test/test_scenario_1.py (hunk 5 above)
Comment (jlmaurer) [reply test/test_scenario_1.py]: @Askaholic fixed.

File: test/test_util.py (hunk 3 above)
Comment (jlmaurer) [reply test/test_util.py:197]: @Askaholic Not sure about this one and the other formatting issues; they ought to be PEP8 compliant (other than line length) as I ran that before the most recent commit (sorry wasn't quite finished with the last push).

File: test/test_scenario_1.py (hunk 5 above)
Comment (Askaholic) [reply test/test_scenario_1.py]: Use the "Resolve conversation" button :)

File: test/test_util.py (hunk 3 above)
Comment (jlmaurer) [reply test/test_util.py:197]: I think the newest update resolved this.

File: tools/RAiDER/models/allowed.py (hunk 4 above)
Comment (jlmaurer) [reply tools/RAiDER/models/allowed.py]: Fixed in newest version.

File: test/test_interpolator.py (hunk 2 above)
Comment (jlmaurer) [reply test/test_interpolator.py]: Fixed with the latest commit

File: test/test_interpolator.py [hunk 8]
```diff
@@ -356,7 +358,8 @@ def test_2d_basic():
     xs = np.array([0, 1])
     ys = np.array([0, 1])
 
-    values = (lambda x, y: x + y)(*np.meshgrid(xs, ys, indexing="ij", sparse=True))
+    values = (lambda x, y: x + y)(*np.meshgrid(xs,
+                                               ys, indexing="ij", sparse=True))
```
Comment (Askaholic) [inline test/test_interpolator.py:362]:
```suggestion
    values = (lambda x, y: x + y)(
        *np.meshgrid(xs, ys, indexing="ij", sparse=True)
    )
```

File: test/test_util.py (hunk 3 above)
Comment (Askaholic) [reply test/test_util.py:197]: PEP compliance is one thing, beautiful code is another. Ultimately, you want both.

File: tools/RAiDER/utilFcns.py [hunk 9]
```diff
@@ -281,9 +287,9 @@ def checkShapes(los, lats, lons, hts):
 
     if not test1 and test2:
         raise ValueError(
-         'I need lats, lons, heights, and los to all be the same shape. ' +
-         'lats had shape {}, lons had shape {}, '.format(lats.shape, lons.shape) +
-         'heights had shape {}, and los was not Zenith'.format(hts.shape))
+            'I need lats, lons, heights, and los to all be the same shape. ' +
```
Comment (Askaholic) [inline tools/RAiDER/utilFcns.py:290]: BTW you can split string literals across multiple lines like this.
```suggestion
            'I need lats, lons, heights, and los to all be the same shape. '
```

Comment (jlmaurer) [reply tools/RAiDER/utilFcns.py:290]: I'll make a note to update throughout, as probably there are other places with this that could all be fixed together.

File: test/test_util.py (hunk 3 above)
Comment (jlmaurer) [reply test/test_util.py:197]: For myself I think the current one looks better...

**Pull request merged**